File: backend/Simpleecom.Shared/Models/Product/Product.cs
namespace Simpleecom.Shared.Models
{
    public class Product : Item
    {
        public Product(
            string name,
            string Description,
            string photoURL,
            double Price,
            int Inventory,
            string Brand,
            int productId
        )
        {
            this.Name = name;
            this.Description = Description;
            this.photoURL = photoURL;
            this.Price = Price;
            this.Inventory = Inventory;
            this.Brand = Brand;
            this.productId = productId;
        }

        public string Name { get; set; }

        public string Description { get; set; } = string.Empty;

        public string photoURL { get; set; } = string.Empty;

        public double Price { get; set; }

        public int Inventory { get; set; }

        public string Brand { get; set; }

        public int productId { get; set; }

        public string GetPartitionKeyValue() => Brand;

    }
}

File: src/frontend/src/components/Products/Product.js
// Product.js
import React from 'react';
import { Link } from 'react-router-dom';

const Product = ({ product, onAddToCart, onBrandLinkClick, onRemoveFromCart, cart }) => {
  if (!product) {
    console.error('Product is undefined');
    return <div>Product is undefined</div>;
  }

  // Use substring to limit the description to the first 100 characters
  const truncatedDescription = product.description ? product.description.substring(0, 100) : '';
  let productName = product.name;
  let productPrice = product.price;
  let productId = product.id
  let brand = product.brand;



return (
  truncatedDescription && (
    <div>
      <img src={product.photoURL} alt={productName} style={{ width: '100%', height: 'auto' }} />
      <h2><Link to={`/product/${productId}`}>{productName}</Link></h2>
      <h3><strong>Brand: </strong><Link to={`/brand/${encodeURIComponent(brand)}`} onClick={() => onBrandLinkClick(brand)}>{brand}</Link></h3>
      <p>{truncatedDescription}...</p>
      <p>${productPrice}</p>
      <button style={{marginBottom: '10px'}} onClick={() => onAddToCart(productId, productName, productPrice)}>Add to Cart</button>
      <button style={{backgroundColor: '#E46A4C'}} onClick={() => onRemoveFromCart(productId, cart)}>Remove from cart</button>
    </div>
  )
);
};

export default React.memo(Product);
